glib-macros/tests/dynamic_enums.rs

```diff
@@ -0,0 +1,494 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::{prelude::*, subclass::prelude::*, Cast};
+
+mod module {
+    use super::*;
+
+    mod imp {
+        use super::*;
+
+        // impl for a type module (must extend `glib::TypeModule` and must implement `glib::TypePlugin`).
+        #[derive(Default)]
+        pub struct MyModule;
+
+        #[glib::object_subclass]
+        impl ObjectSubclass for MyModule {
+            const NAME: &'static str = "MyModule";
+            type Type = super::MyModule;
+            type ParentType = glib::TypeModule;
+            type Interfaces = (glib::TypePlugin,);
+        }
+
+        impl ObjectImpl for MyModule {}
+
+        impl TypePluginImpl for MyModule {}
+
+        impl TypeModuleImpl for MyModule {
+            fn load(&self) -> bool {
+                // registers enums as dynamic types.
+                let my_module = self.obj();
+                let type_module: &glib::TypeModule = my_module.upcast_ref();
+                super::MyModuleEnum::on_implementation_load(type_module)
+                    && super::MyModuleEnumLazy::on_implementation_load(type_module)
```

Comment:
What happens if the user forgets to call these here but then something tries to use the enums?

Reply:
Can we only do that for dynamic enums? It seems like unnecessary code bloat otherwise (all the assertions inserted everywhere with their stack unwinding machinery). Otherwise seems like a good approach.

However, there's also nothing really wrong with having these traits work if the type is not registered yet, is there? Things only have to fail if a `GType` is needed, and that should already lead to assertions. So maybe that's just not necessary at all?

Reply:
Maybe :sweat_smile: I think this behaviour makes more sense, but time will tell us!